backend/vxlan/utils.go

```diff
@@ -151,3 +151,50 @@ func getHostsMap(hosts []metadata.Host) map[string]metadata.Host {
 	logrus.Debugf("hostsMap: %v", hostsMap)
 	return hostsMap
 }
+
+func getLocalRouterInfo(networks []metadata.Network, host metadata.Host, services []metadata.Service) (metadata.Network, metadata.Container, metadata.Service) {
+	var (
+		vxlanNetwork metadata.Network
+		localRouter  metadata.Container
+		selfService  metadata.Service
+	)
+	for _, aNetwork := range networks {
+		if aNetwork.EnvironmentUUID != host.EnvironmentUUID {
+			continue
+		}
+		_, ok := aNetwork.Metadata["cniConfig"].(map[string]interface{})
+		if !ok {
+			continue
+		}
+		vxlanNetwork = aNetwork
+		break
+	}
+
+Loop:
```

Comment:
Personally I am not a fan of labels. You could avoid this if you fetch the service first and then loop on the containers of that service later.